react-catalog/my-app/src/Details.tsx

```diff
@@ -0,0 +1,64 @@
+import { useNavigate, useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { Product, readCatalog } from './lib';
+
+export function Details() {
+  const [items, setItems] = useState<Product[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const { itemId } = useParams();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    async function loadItems(): Promise<void> {
+      try {
+        const values = await readCatalog();
```

Comment:
Ah, this is why the code below is so strange. Instead of calling `readCatalog` you should call `readProduct`, which will give you a single `Product`. Then your code below doesn't need to have that for-loop.